### PR TITLE
Update file.managed arguments for consistency.

### PR DESCRIPTION
**Pull request by abednarik**
As reported in #29483 the mail goal of this is to use show_changes in file state.

### CODE REVIEW

File: salt/states/file.py [hunk 1]
```diff
@@ -1469,15 +1474,30 @@ def managed(name,
...
+        )
+        salt.utils.warn_until('Boron', msg)
+    elif not show_diff and show_diff is not None:
+        show_changes = show_diff
+        msg = (
+            'The \'show_diff\' argument to the file.managed state has been '
+            'deprecated, please use \'show_changes\' instead.'
+        )
+        salt.utils.warn_until('Boron', msg)
+
```
Comment (basepi) [inline salt/states/file.py]: Can we combine this into a single `if show_diff is not None`? I think having two if statements is redundant.

**Comment by basepi** [comment]
One inline comment, but otherwise this looks solid.

**Comment by abednarik** [comment]
Will try to do that.  Thanks.

**Comment by abednarik** [comment]
@basepi  please review if that is what you suggested. I'm pretty sure since i made a totally noob if :)

**Comment by basepi** [comment]
Looks perfect! Thanks!

**Pull request merged**